edgexfoundry/go-mod-bootstrap
# feat(security): Add func to process CORS

Add a http middleware function to process CORS.
Upgrade the http related go module to meet other services.

Signed-off-by: Cloud Tsai <cloudxxx8@gmail.com>







If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?) it's not worth to add unit test for a simple http middleware
- [ ] I have opened a PR for the related docs change (if not, why?) Create document to instruct the CORS settings edgex-docs#608
  <link to docs PR>

## Testing Instructions

1. Add ProcessCORS as a middleware function in the router.go of any service.
2. Modify the configuration to set EnableCORS = true
3. Send a http request to the service and verify the response.
